astropy/stats/lombscargle/_lombscargle.py

```diff
@@ -0,0 +1,255 @@
+            - 'slow': use the O[N^2] pure-python implementation
+            - 'cython': use the O[N^2] cython implementation. This is slightly
+              faster than method='slow', but much more memory efficient.
+            - 'chi2': use the O[N^2] chi2/linear-fitting implementation
+            - 'fastchi2': use the O[N log N] chi2 implementation. Note that this
+              requires evenly-spaced frequencies: by default this will be checked
+              unless ``assume_regular_frequency`` is set to True.
+            - 'scipy': use ``scipy.signal.lombscargle``, which is an O[N^2]
+              implementation written in C. Note that this does not support
+              heteroskedastic errors.
+
+        assume_regular_frequency : bool (optional)
+            if True, assume that the input frequency is of the form
+            freq = f0 + df * np.arange(N). Only referenced if method is 'auto'
+            or 'fast'.
+        normalization : string (optional, default='normalized')
+            Normalization to use for the periodogram. Options are 'normalized'
+            or 'unnormalized'.
+        fit_bias : bool (optional, default=True)
+            if True, include a constant offet as part of the model at each
+            frequency. This can lead to more accurate results, especially in
+            the case of incomplete phase coverage.
+        center_data : bool (optional, default=True)
+            if True, pre-center the data by subtracting the weighted mean of
+            the input data. This is especially important if fit_bias = False
+        method_kwds : dict (optional)
+            additional keywords to pass to the lomb-scargle method
+
+        Returns
+        -------
+        power : ndarray
+            The Lomb-Scargle power at the specified frequency
+        """
+        if frequency is None:
+            raise ValueError("Must supply a valid frequency. If you would "
+                             "like an automatic frequency grid, use the "
+                             "autopower() method.")
+        return lombscargle(self.t, self.y, self.dy,
+                           frequency=frequency,
+                           center_data=self.center_data,
+                           fit_bias=self.fit_bias,
+                           normalization=normalization,
+                           method=method, method_kwds=method_kwds,
+                           assume_regular_frequency=assume_regular_frequency)
+
+    def model(self, t, frequency):
+        """Compute the Lomb-Scargle model at the given frequency
+
+        Parameters
+        ----------
+        t : array_like, length n_samples
+            times at which to compute the model
+        frequency : float
+            the frequency for the model
+
+        Returns
+        -------
+        y : np.ndarray, length n_samples
+            The model fit corresponding to the input times
+        """
+        # TODO: handle units correctly
```

Comment:
No, this function actually does not handle units yet. It's not really an essential feature though...